### DISCUSSION TITLE
Bagaimana mengatur limit karakter pada meta description di Mecha-CMS?

### DISCUSSION THREAD

**Discussion by bungfrangki** [mecha-cms/mecha#279] [Q&A] [answered]
pada berkas _enter.php_ saya mendapati **meta-description** ditulis seperti ini:

```
<?php if ($w = w($page->description ?? $site->description ?? "")): ?>
      <meta content="<?= $w; ?>" name="description">
    <?php endif; ?>
```

bagaimana cara mengatur batas karakter pada meta description tersebut?
misalnya, saya hanya ingin menampilkan 160 karakter pada meta description. thx

**Comment by taufik-nurrohman** [comment, marked as answer]
Kalau sekedar untuk membatasi jumlah bisa pakai `substr()`, kalau mau menambahkan titik tiga di akhir perpotongan bisa pakai [`To::description()`](https://mecha-cms.com/reference/3.x/extension/page/object/to/description):

~~~ php
<meta content="<?= substr($w, 0, 160); ?>" name="description">
~~~

~~~ php
<meta content="<?= To::description($w, 160); ?>" name="description">
~~~

**Reply by bungfrangki** [reply]
mantap mas, berhasil.
saya coba terapkan juga pada twitter card dan facebook open graph,
saya terapkan di kode ini: `$page->description` misalnya seperti ini `\w(substr($page->description, 0, 160))` jumlah karakter outputnyanya jadi gak sinkron sama kode `substr($w, 0, 160)` bila kondisinya pada _description_ terdapat link.

Untuk kode `\w(substr($page->description, 0, 160))` output karakternya terlihat lebih sedikit.

**Reply by bungfrangki** [reply]
ah maaf mas,
tapi kalau pake ` w(To::description($w, 160))` hasil outputnya jadi sama persis. jumlah karakternya ngikut sama yang `substr($w, 0, 160)`. Cuma masalahnya sy blum berhasil menerapkan `To::description()` dengan kode seperti ini `\w(To::description($page->description, 160))` di twitter card dan facebook open graph.

**Reply by bungfrangki** [reply]
maaf mas, saya udah ketemu solusinya.
saya keliru menerapkan ini `\w(substr($page->description, 0, 160))` di ekstensi twitter card dan facebook open graph.
mestinya dilakukan pada variabel `$description` yang disini: `['meta', false, ['content' => $description, 'name' => 'twitter:description']]`
hehehe

**Reply by taufik-nurrohman** [reply]
Emang ada batasan harus 160 karakter apa mas @bungfrangki? Minta referensinya dong.

**Reply by bungfrangki** [reply]
sebetulnya memang tak ada batasan maksimal minimalnya mas, cuma sebatas pengamatan saja terhadap best practice berbagai tools seo checker macam Moz, rata2 menggunakan angka maksimal 160 karakter untuk meta description.

Kalau plugin SEO macam Yoast, bahkan lebih kurang dari itu, hanya up to 120-155 karakter.

sebetulnya memberi limit karakter di meta description ini termasuk kategori penting gak penting si. Namun meski dikatakan tidak begitu penting, sy tetap perlu concern terhadap duplicate content. Biasanya kalau meta description kepanjangan, tampilan snippet di SERP kepotong-potong gak karuan, yg hasil tampilan snippetnya bisa beda-beda di tiap user.

https://moz.com/learn/seo/meta-description/
https://yoast.com/meta-descriptions/

**Reply by taufik-nurrohman** [reply]
Oke deh.
